controllers/loadtest_controller.go

```diff
@@ -125,6 +125,70 @@ func (r *LoadTestReconciler) Reconcile(req ctrl.Request) (ctrl.Result, error) {
 	return ctrl.Result{}, nil
 }
 
+// CheckMissingPods attempts to check if any pods is missing for current
+// LoadTest.
+func CheckMissingPods(currentLoadTest *grpcv1.LoadTest, allRunningPods *corev1.PodList) []*grpcv1.Component {
+
+	//Creat maps for Clients and Servers for current currentLoadTest
+	CurrentLoadTestClientPodMap := make(map[string]*grpcv1.Component)
+	CurrentLoadTestServerPodMap := make(map[string]*grpcv1.Component)
+	foundDriver := false
+	var missingComponents []*grpcv1.Component
+
+	//populate the client and server map
+	for i := 0; i < len(currentLoadTest.Spec.Clients); i++ {
+		CurrentLoadTestClientPodMap[*currentLoadTest.Spec.Clients[i].Name] = &currentLoadTest.Spec.Clients[i].Component
+	}
+	for i := 0; i < len(currentLoadTest.Spec.Servers); i++ {
+		CurrentLoadTestServerPodMap[*currentLoadTest.Spec.Servers[i].Name] = &currentLoadTest.Spec.Servers[i].Component
+	}
+
+	// Iterate through all existing pod, once found remove the pod from the map
+	//The leftover entries would be the ones are missing
+	for _, eachPod := range allRunningPods.Items {
+		loadTestLabelForEachPod := eachPod.Labels[defaults.LoadTestLabel]
+		roleLabelForEachPod := eachPod.Labels[defaults.RoleLabel]
+		componentNameLabelForEachPod := eachPod.Labels[defaults.ComponentNameLabel]
+
+		// ignore pods don't belong to current loadtest
+		if loadTestLabelForEachPod != currentLoadTest.Name {
+			continue
+		}
+		if roleLabelForEachPod == "driver" {
```

Comment:
Lets not rely on string literals in-case we change the component role at some point. 

```suggestion
		if roleLabelForEachPod == defaults.DriverRole {
```

Reply:
Agreed! Should minimize hardcoded names or labels.